pallets/did/src/mock.rs

```diff
@@ -17,400 +17,433 @@
+		TestCtypeHash::from_slice(&[1u8; 32])
+	}
+	pub(crate) fn get_authentication_key_call() -> Call {
+		Call::Ctype(ctype::Call::add(get_authentication_key_test_input()))
+	}
+	pub(crate) fn get_delegation_key_test_input() -> TestCtypeHash {
+		TestCtypeHash::from_slice(&[2u8; 32])
+	}
+	pub(crate) fn get_delegation_key_call() -> Call {
+		Call::Ctype(ctype::Call::add(get_delegation_key_test_input()))
+	}
+	pub(crate) fn get_none_key_test_input() -> TestCtypeHash {
+		TestCtypeHash::from_slice(&[3u8; 32])
+	}
+	pub(crate) fn get_none_key_call() -> Call {
+		Call::Ctype(ctype::Call::add(get_none_key_test_input()))
 	}
-}
 
-#[allow(dead_code)]
-pub fn initialize_logger() {
-	env_logger::builder().is_test(true).try_init();
-}
+	impl did::DeriveDidCallAuthorizationVerificationKeyRelationship for Call {
+		fn derive_verification_key_relationship(&self) -> Option<did::DidVerificationKeyRelationship> {
+			if *self == get_attestation_key_call() {
+				Some(did::DidVerificationKeyRelationship::AssertionMethod)
+			} else if *self == get_authentication_key_call() {
+				Some(did::DidVerificationKeyRelationship::Authentication)
+			} else if *self == get_delegation_key_call() {
+				Some(did::DidVerificationKeyRelationship::CapabilityDelegation)
+			} else {
+				#[cfg(feature = "runtime-benchmarks")]
+				if *self == Self::get_call_for_did_call_benchmark() {
+					// Always require an authentication key to dispatch calls during benchmarking
+					return Some(did::DidVerificationKeyRelationship::Authentication);
+				}
+				None
+			}
+		}
 
-#[derive(Clone)]
-pub struct ExtBuilder {
-	dids_stored: Vec<(TestDidIdentifier, did::DidDetails<Test>)>,
-}
+		// Always return a System::remark() extrinsic call
+		#[cfg(feature = "runtime-benchmarks")]
+		fn get_call_for_did_call_benchmark() -> Self {
+			Call::System(frame_system::Call::remark(vec![]))
+		}
+	}
 
-impl Default for ExtBuilder {
-	fn default() -> Self {
-		Self { dids_stored: vec![] }
+	pub fn generate_test_did_call(
+		verification_key_required: did::DidVerificationKeyRelationship,
+		caller: TestDidIdentifier,
+	) -> did::DidAuthorizedCallOperationWithVerificationRelationship<Test> {
+		let call = match verification_key_required {
+			DidVerificationKeyRelationship::AssertionMethod => get_attestation_key_call(),
+			DidVerificationKeyRelationship::Authentication => get_authentication_key_call(),
+			DidVerificationKeyRelationship::CapabilityDelegation => get_delegation_key_call(),
+			_ => get_none_key_call(),
+		};
+		did::DidAuthorizedCallOperationWithVerificationRelationship {
+			operation: did::DidAuthorizedCallOperation {
+				did: caller,
+				call,
+				tx_counter: 1u64,
+			},
+			verification_key_relationship: verification_key_required,
+		}
 	}
-}
 
-impl ExtBuilder {
-	pub fn with_dids(mut self, dids: Vec<(TestDidIdentifier, did::DidDetails<Test>)>) -> Self {
-		self.dids_stored = dids;
-		self
+	#[allow(unused_must_use)]
```

Comment:
Out of the scope of this PR, but we might also want to move all these utility functions into their own crate, probably.